### PR TITLE
Update Qualys Webapp Parser to use Unique ID from tool

### PR DESCRIPTION
**Pull request by Hijerboa**
This PR fixes several issues with the Qualys Webapp parser

1. As of the latest dev update, which uses the new finding import structure, the qualys webapp parser was broken, as it did not directly assign the test field in a finding. This fixes that.
2. Previously, the parser would group all vulnerabilities with a QID under one finding. This means that even if there were 1000 reflected XSS vulns, DD would only display one finding. With this parser change, each vulnerability is considered a finding. Because of this, the number of findings in Defect Dojo now lines up with the number of vulnerabilities reported by Qualys WAS.
3. Because all findings with one QID were grouped under a single finding, the request response pairs listed in DD were not accurate. The finding would list one request/response pair, no matter how many vulnerabilities were listed under it. Because each vulnerability now has it's own finding, this is no longer an issue.

### CODE REVIEW

**Comment by Maffooch** [comment]
Hi @Hijerboa, the parser was originally written so that all findings under one QID would be aggregated under one finding. This idea was supported by the endpoint statuses that allowed partial remediation of a finding by mitigating individual endpoints. 

There is also dedicated metrics pages that shows number of endpoints as well as number of findings to show the difference in these numbers.

If having multiple findings for identical issues works better for your flow, then please hide this feature behind a feature flag in `settings.dist.py` so it does not break other users work flow. Also please add some notes in the documentation so others users will know about it.

Thanks!

File: dojo/tools/qualys_webapp/parser.py [hunk 1]
```diff
@@ -207,28 +227,28 @@ def get_info_item(info_gathered, finding):
...
+        qid = int(finding.vuln_id_from_tool)
         if qid in g_qid_list:
             index = g_qid_list.index(qid)
             finding = get_glossary_item(glossary[index], finding, True, enable_weakness)
         if qid in ig_qid_list:
             index = ig_qid_list.index(qid)
-            findings[qid] = get_info_item(info_gathered[index], finding)
-
+            findings[unique_id] = get_info_item(info_gathered[index], finding)
     return findings
```
Comment (damiencarol) [inline dojo/tools/qualys_webapp/parser.py:251]:
```suggestion


```

**Comment by Hijerboa** [comment]
Hi @Maffooch I've updated the parser so that it now will only create a unique finding for each unique ID if a feature flag that I created in `settings.dist.py` has been set to True. 

Additionally, this change also prevents an error on upload of a qualys webapp scan with the latest version of dev's changes to parser import structure.

File: dojo/settings/settings.dist.py [hunk 2]
```diff
@@ -811,6 +811,7 @@ def generate_url(scheme, double_slashes, user, password, host, port, path, param
     'Whitesource Scan': ['title', 'severity', 'description'],
     'ZAP Scan': ['title', 'cwe', 'severity'],
     'Qualys Scan': ['title', 'severity'],
+    'Qualys Webapp Scan': ['title', 'unique_id_from_tool', 'endpoints'],
```
[the remark this answers is not in the document]
Comment (Hijerboa) [reply dojo/settings/settings.dist.py]: Hi @damiencarol I've updated the hash code to not make use of the endpoints

**Review by Maffooch — APPROVED** [review]

**Review by madchap — APPROVED** [review]

**Pull request merged**